controllers/worker_controller.go

```diff
@@ -46,17 +65,186 @@ type WorkerReconciler struct {
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
-func (r *WorkerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+func (r *WorkerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, err error) {
+	log := ctrl.LoggerFrom(ctx)
+	log.V(1).Info("Start Worker Reconcile")
+	worker := &arcadiav1alpha1.Worker{}
+	if err := r.Get(ctx, req.NamespacedName, worker); err != nil {
+		// There's no need to requeue if the resource no longer exists.
+		// Otherwise, we'll be requeued implicitly because we return an error.
+		log.V(1).Info("Failed to get Worker")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	log.V(1).Info("Get Worker instance")
+
+	// Add a finalizer.Then, we can define some operations which should
+	// occur before the Worker to be deleted.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers
+	if newAdded := controllerutil.AddFinalizer(worker, arcadiav1alpha1.Finalizer); newAdded {
+		log.Info("Try to add Finalizer for Worker")
+		if err = r.Update(ctx, worker); err != nil {
+			log.Error(err, "Failed to update Worker to add finalizer, will try again later")
+			return ctrl.Result{Requeue: true}, err
+		}
+		log.Info("Adding Finalizer for Worker done")
+		return ctrl.Result{Requeue: true}, nil
```

Comment:
**Suggestion:** Consider handling the error returned by `controllerutil.AddFinalizer(worker, arcadiav1alpha1.Finalizer)`. Currently, the error is ignored which might lead to unexpected behavior if the finalizer cannot be added.
```suggestion
 newAdded, err := controllerutil.AddFinalizer(worker, arcadiav1alpha1.Finalizer)
 if err != nil {
   log.Error(err, "Failed to add finalizer for Worker")
   return ctrl.Result{}, err
 }
 if newAdded {
   log.Info("Try to add Finalizer for Worker")
   if err = r.Update(ctx, worker); err != nil {
     log.Error(err, "Failed to update Worker to add finalizer, will try again later")
     return ctrl.Result{Requeue: true}, err
   }
   log.Info("Adding Finalizer for Worker done")
   return ctrl.Result{Requeue: true}, nil
 }
```